src/integration/pushResult.js

```diff
@@ -73,13 +103,32 @@ function addCommit(src, msg){
 
 //pushing to remote repo
 function push(branch, src){
-
   return (cb) => {
-
     git.push('origin', branch, {cwd: src}, (err) => {
       cb(err);
     });
   };
 }
 
 module.exports = pushResult;
+
+/**
+ * Function resposible for moving files between two directories in order to backup them or restore them
+ * @param {Boolean} [from] - responsible for choosing the src folder: `${item}/*` or `./tmp/backup/${path.normalize(item)}/*`
+ * @param {Boolean} [to] - responsible for choosing the destination folder: `${item}/` or `./tmp/backup/${path.normalize(item)}/`
+ * @param {Function} [cb] - callback for asynchronous operation
+*/
+
+function _backup(from, to, cb){
+  const data = fs.readFileSync('./tmp/notClonedRepositories.json', 'utf-8');
+  const array = data.toString().split(',');
+  let itemsProcessed = 0;
+
+  array.forEach((item) => {
```

Comment:
My idea:
build array with async operations and invoke at series/parallel. 
PSEUDO CODE : 
const arrOfTask = []

array.forEach((item) => {
   arrOfTask.push(copier.copyFilesAsync(someParams))
}

async.series(arrOfTask, cb)